### ISSUE TITLE
Allow the user to set the loop tied to a CoroutineMock

### ISSUE THREAD

**Issue by Martiusweb** [Martiusweb/asynctest#80]
Now that CoroutineMock relies on the loop for some features, we should let the user specify which loop shall be used.

    CoroutineMock(loop=None)  # default, in this case, get_event_loop() is called lazily when a loop is required

    CoroutineMock(loop=user_defined_loop)  # This loop is used instead of the one returned by get_event_loop()

**Comment by Kentzo** [comment]
Makes sense. But we should do our best to make it work without specifying the loop.

**Comment by RemiCardona** [comment]
Yeah, I hit that one. It breaks TestCase classes where `forbid_get_event_loop` is enabled. Bit of a bummer.
